templates/workspace_services/guacamole/terraform/web_app.tf

```diff
@@ -35,15 +35,15 @@ resource "azurerm_app_service" "guacamole" {
     GUAC_DRIVE_NAME       = "${var.guac_drive_name}"
     GUAC_DRIVE_PATH       = "${var.guac_drive_path}"
     GUAC_DISABLE_DOWNLOAD = "${var.guac_disable_download}"
-    AUDIENCE              = "${var.openid_client_id}"
+    AUDIENCE              = "${var.ws_client_id}"
     ISSUER                = local.issuer
 
-    OPENID_AUTHORIZATION_ENDPOINT = "https://login.microsoftonline.com/${local.aad_tenant_id}/oauth2/v2.0/authorize"
-    OPENID_JWKS_ENDPOINT          = "https://login.microsoftonline.com/${local.aad_tenant_id}/discovery/v2.0/keys"
-    OPENID_ISSUER                 = "https://login.microsoftonline.com/${local.aad_tenant_id}/v2.0"
-    OPENID_CLIENT_ID              = "${var.openid_client_id}"
-    OPENID_REDIRECT_URI           = "https://${local.webapp_name}.azurewebsites.net/guacamole/"
-    OPENID_USERNAME_CLAIM_TYPE    = "email"
+    OAUTH2_PROXY_CLIENT_ID       = "${var.ws_client_id}"
+    OAUTH2_PROXY_CLIENT_SECRET   = "${var.ws_client_secret}"
+    OAUTH2_PROXY_REDIRECT_URI    = "https://${local.webapp_name}.azurewebsites.net/oauth2/callback"
+    OAUTH2_PROXY_EMAIL_DOMAIN    = "\"*\"" # oauth proxy will allow all email domains only when the value is "*"
+    OAUTH2_PROXY_OIDC_ISSUER_URL = "https://login.microsoftonline.com/${local.aad_tenant_id}/v2.0"
+    OAUTH2_PROXY_JWKS_ENDPOINT   = "https://login.microsoftonline.com/${local.aad_tenant_id}/discovery/v2.0/keys"
```

Comment:
There are instances where these will need to be customized. 

For example if access to the web app is via a custom FQDN, or a different OIDC provider is used. Again might not be one to handle in this PR, but we should create issues to track.